src/modules/ekf2/ekf2_main.cpp

```diff
@@ -506,18 +504,14 @@ void Ekf2::task_main()
 			continue;
 		}
 
-		if (fds[1].revents & POLLIN) {
+		bool params_updated = false;
+		orb_check(params_sub, &params_updated);
+
+		if (params_updated) {
 			// read from param to clear updated flag
 			struct parameter_update_s update;
 			orb_copy(ORB_ID(parameter_update), params_sub, &update);
 			updateParams();
-
-			// fetch sensor data in next loop
-			continue;
-
-		} else if (!(fds[0].revents & POLLIN)) {
-			// no new data
-			continue;
```

Comment:
Alright, that's definitely the safe thing to do. I didn't think we ever set anything but POLLIN.
We should push these patterns into the Subscription wrapper or something.